### PR TITLE
feat(interview-prep): Converting and Transfering Rosetta problems

### PR DESCRIPTION
**Pull request by NitronR**
Checklist:

- [ ] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ ] My pull request targets the `master` branch of freeCodeCamp.
- [ ] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

### CODE REVIEW

**Comment by raisedadead** [comment]
The CI is broken due to unrelated changes to the PR. We will rebase this soon.

**Comment by NitronR** [comment]
@raisedadead Okay.

**Comment by NitronR** [comment]
@raisedadead Thanks for the help.
@scissorsneedfoodtoo Here is the next set of problems. Can you please review this?

**Comment by scissorsneedfoodtoo** [comment]
@NitronR, thanks for your patience. Sure, I'll look at this now.

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/ludic-numbers.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/ludic-numbers.md:10]: Let's link to the secure versions of each page, and use `target="_blank"`:

`<a href="https://oeis.org/wiki/Ludic_numbers" target="_blank">Ludic numbers</a> are related to prime numbers as they are generated by a sieve quite like the <a href="https://rosettacode.org/wiki/Sieve_of_Eratosthenes" target="_blank">Sieve of Eratosthenes</a> is used to generate prime numbers.`

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/ludic-numbers.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/ludic-numbers.md:27]: Let's add some spacing and margins to make things a little more readable here:

```
<code style="margin-left: 2em;"><span style="color:blue;font-weight:bold">2</span> 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 ...</code>
(Loop)
<ul>
  <li>Take the first member of the resultant array as the next ludic number <span style="color:blue;font-weight:bold">2</span>.</li>
  <li>Remove every <strong>2<sup>nd</sup></strong> indexed item from the array (including the first).</li>
  <code style="margin-left: 2em;"><span style="color:blue;font-weight:bold;"><s>2</s></span> 3 <s>4</s> 5 <s>6</s> 7 <s>8</s> 9 <s>10</s> 11 <s>12</s> 13 <s>14</s> 15 <s>16</s> 17 <s>18</s> 19 <s>20</s> 21 <s>22</s> 23 <s>24</s> 25 <s>26</s> ...</code>
</ul>
<ul>
  <li>(Unrolling a few loops...)</li>
  <li>Take the first member of the resultant array as the next ludic number <span style="color:blue;font-weight:bold">3</span>.</li>
  <li>Remove every <strong>3<sup>rd</sup></strong> indexed item from the array (including the first).</li>
  <code style="margin-left: 2em;"><span style="color:blue;font-weight:bold"><s>3</s></span> 5 7 <s>9</s> 11 13 <s>15</s> 17 19 <s>21</s> 23 25 <s>27</s> 29 31 <s>33</s> 35 37 <s>39</s> 41 43 <s>45</s> 47 49 <s>51</s> ...</code></span>
</ul>
<ul>
  <li>Take the first member of the resultant array as the next ludic number <span style="color:blue;font-weight:bold">5</span>.</li>
  <li>Remove every <strong>5<sup>th</sup></strong> indexed item from the array (including the first).</li>
  <code style="margin-left: 2em;"><span style="color:blue;font-weight:bold"><s>5</s></span> 7 11 13 17 <s>19</s> 23 25 29 31 <s>35</s> 37 41 43 47 <s>49</s> 53 55 59 61 <s>65</s> 67 71 73 77 ...</code></span>
</ul>
<ul>
  <li>Take the first member of the resultant array as the next ludic number <span style="color:blue;font-weight:bold">7</span>.</li>
  <li>Remove every <strong>7<sup>th</sup></strong> indexed item from the array (including the first).</li>
  <code style="margin-left: 2em;"><span style="color:blue;font-weight:bold"><s>7</s></span> 11 13 17 23 25 29 <s>31</s> 37 41 43 47 53 55 <s>59</s> 61 67 71 73 77 83 <s>85</s> 89 91 97 ...</code>
</ul>
<ul>
  <li><big><b> ... </b></big></li>
  <li>Take the first member of the current array as the next ludic number <span style="color:blue;font-weight:bold">L</span>.</li>
  <li>Remove every <strong>L<sup>th</sup></strong> indexed item from the array (including the first).</li>
  <li><big><b> ... </b></big></li>
</ul>
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/ludic-numbers.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/ludic-numbers.md:43]: `testString: assert(typeof ludic === 'function');`

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/luhn-test-of-credit-card-numbers.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/luhn-test-of-credit-card-numbers.md:10]: The <a href="https://en.wikipedia.org/wiki/Luhn algorithm" target="_blank">Luhn test</a> is used by some credit card companies to distinguish valid credit card numbers from what could be a random selection of digits.

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/luhn-test-of-credit-card-numbers.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/luhn-test-of-credit-card-numbers.md:17]: We can make this more readable and fix the lists:

```
<ol>
  <li> Reverse the order of the digits in the number.</li>
  <li> Take the first, third, ... and every other odd digit in the reversed digits and sum them to form the partial sum s1</li>
  <li> Taking the second, fourth ... and every other even digit in the reversed digits:</li>
    <ol>
      <li>Multiply each digit by two and sum the digits if the answer is greater than nine to form partial sums for the even digits.</li> 
      <li>Sum the partial sums of the even digits to form s2.</li>
    </ol>
  <li>If s1 + s2 ends in zero then the original number is in the form of a valid credit card number as verified by the Luhn test.</li>
</ol>

```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/luhn-test-of-credit-card-numbers.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/luhn-test-of-credit-card-numbers.md:30]: Seems like a code block and spacing are missing here:

```
For example, if the trial number is 49927398716:

```
Reverse the digits:
  61789372994
Sum the odd digits:
  6 + 7 + 9 + 7 + 9 + 4 = 42 = s1
The even digits:
    1, 8, 3, 2, 9
  Two times each even digit:
    2, 16, 6, 4, 18
  Sum the digits of each multiplication:
    2, 7, 6, 4, 9
  Sum the last:
    2 + 7 + 6 + 4 + 9 = 28 = s2

s1 + s2 = 70 which ends in zero which means that 49927398716 passes the Luhn test.
```
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/luhn-test-of-credit-card-numbers.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/luhn-test-of-credit-card-numbers.md:46]: It probably doesn't really matter for this kind of test, but perhaps this could use the strict comparison operator:

`testString: assert(typeof luhnTest === 'function');`

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/lychrel-numbers.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/lychrel-numbers.md:37]: Had to play with the spacing here to get the code blocks and space between paragraphs to render properly:

```
<ol>
  <li>Take an integer <code>n</code>, greater than zero.</li>
  <li>Form the next <code>n</code> of its series by reversing the digits of the current n and adding the result to the current <code>n</code>.</li>
  <li>Stop when <code>n</code> becomes palindromic - i.e. the digits of <code>n</code> in reverse order == <code>n</code>.</li>
</ol>

The above recurrence relation when applied to most starting numbers `n` = 1, 2, ... terminates in a palindrome quite quickly.

For example if `n`<sub>0</sub> = 12 we get:
```
12
12 + 21 = 33,  a palindrome!
```

And if `n`<sub>0</sub> = 55 we get:
```
55
55 + 55 = 110
110 + 011 = 121,  a palindrome!
```
Notice that the check for a palindrome happens <i>after</i> an addition.

Some starting numbers seem to go on forever; the recurrence relation for 196 has been calculated for millions of repetitions forming numbers with millions of digits, without forming a palindrome. These numbers that do not end in a palindrome are called <b>Lychrel numbers</b>.

For the purposes of this task a Lychrel number is any starting number that does not form a palindrome within 500 (or more) iterations.

<strong>Seed and related Lychrel numbers:</strong>

Any integer produced in the sequence of a Lychrel number is also a Lychrel number.

In general, any sequence from one Lychrel number <i>might</i> converge to join the sequence from a prior Lychrel number candidate; for example the sequences for the numbers 196 and then 689 begin:
```
    196
    196 + 691 = 887
    887 + 788 = 1675
    1675 + 5761 = 7436
    7436 + 6347 = 13783
    13783 + 38731 = 52514
    52514 + 41525 = 94039
    ...
    689
    689 + 986 = 1675
    1675 + 5761 = 7436
    ...
```
So we see that the sequence starting with 689 converges to, and continues with the same numbers as that for 196.

Because of this we can further split the Lychrel numbers into true <b>Seed</b> Lychrel number candidates, and <b>Related</b> numbers that produce no palindromes but have integers in their sequence seen as part of the sequence generated from a lower Lychrel number.
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/lzw-compression.md
Comment (scissorsneedfoodtoo) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/lzw-compression.md:10]:
```
You can read a complete description of it in the  <a href="https://en.wikipedia.org/wiki/Lempel-Ziv-Welch" target="_blank">Wikipedia article</a>  on the subject. It was patented, but it entered the public domain in 2004.
```

**Comment by scissorsneedfoodtoo** [comment]
The multiple code blocks in the comments would have been difficult to figure out, so I hope you don't mind that I pushed my changes to your branch.

Everything LGTM. We just need one more review before merging this. Could you take a look if you have some time @thecodingaviator?

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/lychrel-numbers.md
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/lychrel-numbers.md:12]:
```suggestion
  <li>Stop when <code>n</code> becomes palindromic - i.e. the digits of <code>n</code> in reverse order == <code>n</code>.</li>
```

File: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/lychrel-numbers.md
Comment (thecodingaviator) [inline curriculum/challenges/english/08-coding-interview-prep/rosetta-code/lychrel-numbers.md:68]:
```suggestion
Write a function that takes a number as a parameter. Return true if the number is a Lynchrel number. Otherwise, return false. Remember that the iteration limit is 500.
```

**Review by thecodingaviator — CHANGES_REQUESTED** [review]

**Comment by NitronR** [comment]
@scissorsneedfoodtoo Thank you for the review and fixing the problems.
@thecodingaviator Thank you for reviewing this, I have done the changes. Please let me know if anything else can be improved.

**Review by thecodingaviator — APPROVED** [review]

**Comment by thecodingaviator** [comment]
@NitronR just a little FYI: You can click add to batch and commit everything from there. Saves time.

**Review by scissorsneedfoodtoo — APPROVED** [review]

**Comment by scissorsneedfoodtoo** [comment]
Thanks for more amazing ports @NitronR, and thank you @thecodingaviator for your thorough review. Everything LGTM :+1: :+1:

**Comment by moT01** [comment]
I'm wondering if we should try and make sure to add guide pages and `forumTopicId`'s when we merge these.

**Comment by NitronR** [comment]
@moT01 Sure. How can I create those?

**Pull request merged**
